packages/react/src/components/DataTable/DataTable.tsx

```diff
@@ -621,18 +620,17 @@ class DataTable<RowType, ColTypes extends any[]> extends React.Component<
    * Gets the props associated with selection for a header or a row, where
    * applicable. Most often used to indicate selection status of the table or
    * for a specific row.
-   *
-   * @param {object} [row] an optional row that we want to access the props for
-   * @param {Function} row.onClick
-   * @param {object} row.row
-   * @returns {object}
    */
   getSelectionProps = (
-    { onClick, row, ...rest } = {} as {
-      onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
-      row: DataTableRow<ColTypes>;
-      [key: string]: unknown;
-    }
+    // TODO: `row` is required per the type definition. Why does the default
+    // value not include it? The fact that it lacks it makes it seem like it's
+    // unnecessary. Further, there's an `if` statement that checks for it below.
+    // Again, why is that necessary if it's required?
```

Comment:
im thinking row should not be required

Reply:
I updated the `TODO` to target a change at https://github.com/carbon-design-system/carbon/issues/19177.